### DISCUSSION TITLE
Modern SwiftUI: Standups App

### DISCUSSION THREAD

**Discussion by BryanJBryce** [pointfreeco/episode-code-samples#141] [Q&A]
`StandupsList` stores its model in an `ObservedObject`there isn't a `StateObject` further up the line, the model is just passed in as a parameter into the `View`'s init. From what I understand if nothing retains the `ObservableObject` a new instance is created every time that view redraws its body. Was this intended behavior or left as an exercise to the reader?

**Comment by mbrandonw** [comment]
Hi @BryanJBryce, the `StandupsList` view holds onto its observable object like so:

```swift
struct StandupsList: View {
  @ObservedObject var model: StandupsListModel
  …
}
```

…which makes the parent responsible for providing the model. So it will not be recreated unless the parent decides to recreate it.

And the parent is the [app entry point](https://github.com/pointfreeco/episode-code-samples/blob/521eb9b1943b615942939a7fb48fd1899075282c/0220-modern-swiftui-pt7/Standups/Standups/StandupsApp.swift#L13):

```swift
@main
struct StandupsApp: App {
  var body: some Scene {
    WindowGroup {
      StandupsList(
        model: StandupsListModel()
      )
    }
  }
}
```

And that shows that only a single model is created for the entire duration of the app.

Also I am going to move this to a discussion since it isn't really an issue with the code sample.

**Comment by BryanJBryce** [comment]
@mbrandonw What's confusing to me is that there is never a `@StateObject` which I understood was required for basically retaining the state so that when views are redrawn that the data is retained.

**Reply by BryanJBryce** [reply]
I get lots of issues with the views flickering and resetting when I do this, I don't see that in your app, must be where I'm storing the initial one

**Reply by mbrandonw** [reply]
> @mbrandonw What's confusing to me is that there is never a `@StateObject` which I understood was required for basically retaining the state so that when views are redrawn that the data is retained.

That's not true. `@StateObject` is only necessary if the view owns the lifecycle of the object. If the object is being passed in from the parent (as is the case with the Standups app), then it is correct to use `@ObservedObject`.
